packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -282,7 +369,17 @@ export class AiAgentService {
             projectUuid,
         });
 
-        return agents;
+        // Filter agents based on group access
+        const agentsWithAccess = await Promise.all(
+            agents.map(async (agent) => {
```

Comment:
you can filter directly here, instead of down bellow

Reply:
smart

Reply:
looks great, sorry should've marked it as a nit